HARK/ConsumptionSaving/ConsPortfolioModel.py

```diff
@@ -822,6 +824,74 @@ def make_ShareFuncAdj(self):
         Construct the risky share function when the agent can adjust
         """
 
+        # Share function on aGrid
+
+        if self.zero_bound:
+            aNrm_temp = np.append(0.0, self.aNrmGrid)
+            share_temp = np.append(self.ShareLimit, self.Share_now)
+        else:
+            aNrm_temp = self.aNrmGrid
+            share_temp = self.Share_now
+
+        self.EndOfPrdShareFunc = LinearInterp(
+            aNrm_temp, share_temp, intercept_limit=self.ShareLimit, slope_limit=0.0
+        )
+
+        # alternative share functions from linear approximation
+
+        # get next period's consumption and share function
+        cFunc_next = self.solution_next.cFuncAdj
+        sFunc_next = self.solution_next.EndOfPrdShareFunc
+
+        def premium(shock):
+            """
+            Used to evaluate mean and variance of equity premium.
+            """
+            r_diff = shock - self.Rfree
+
+            return r_diff, r_diff ** 2
+
+        prem_mean, prem_var = calc_expectation(self.RiskyDstn, premium)
```

Comment:
@alanlujan91 not sure if this is a bug given your notation, but I think that what your are calling `prem_var` is not the variance of the premium, but rather E[premium^2].

Reply:
It would be interesting to see what happens when you make the maximum amount of assets something much larger, like 2000 or 10000 or something.  Out that far, the consumption function is essentially linear, so the error coming from the approximation's assumption that $c^{''}$ is zero should be inconsequential.  If the orange does converge to the blue, that would suggest that maybe using the 2nd order Taylor approximation would make the results considerably closer for small values of wealth.  If orange does not converge to blue, I'd guess that there's a bug somewhere.

Reply:
@Mv77 you're right, I have called this prem_sqrd instead.